server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupInfo.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.engine.storagegroup;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.apache.iotdb.db.conf.IoTDBDescriptor;
+import org.apache.iotdb.db.rescon.SystemInfo;
+
+/**
+ * The storageGroupInfo records the total memory cost of the Storage Group.
+ */
+public class StorageGroupInfo {
+
+  private StorageGroupProcessor storageGroupProcessor;
+
+  /**
+   * The total memory cost of the unsealed TsFileResources in this SG
+   */
+  private AtomicLong unsealedResourceMemCost;
+
+  /**
+   * The total memtable memory cost in this SG
+   */
+  private AtomicLong memTableCost;
+
+  /**
+   * The total memory cost of ChunkMetadata in this SG
+   */
+  private AtomicLong chunkMetadataMemCost;
+
+  /**
+   * The total memory cost of WALs in this SG
+   */
+  private AtomicLong walMemCost;
```

Comment:
do we need four AtomicLongs?

Reply:
Yes, it is not necessary to use four costs here.  This four AtomicLongs can be merged into to one.